modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp

```diff
@@ -763,7 +765,14 @@ renderResponse.setTitle(headerTitle);
 				titleElement.value = fileFileName.replace(/\.[^.]*$/, '');
 			}
 
-			if (fileNameElement && !fileNameElement.value) {
+			var autoChangeFileName = document.getElementById(
+				'<portlet:namespace />changeFileName'
+			);
+
+			if (
+				fileNameElement &&
+				(!fileNameElement.value || autoChangeFileName.checked)
```

Comment:
I had doubts about this `||`, if it shouldn't be a `&&`. But I guess it is correct for this LPP, because when uploading a different file, `fileNameElement.value` has already a value that we want to overwrite while the checkbox is checked! 👏